### PR TITLE
Create Contributing.MD

### PR DESCRIPTION
**Pull request by mmiddle5**

### CODE REVIEW

File: Contributing.MD [hunk 1]
```diff
@@ -0,0 +1,30 @@
+Contributing to README Template
+
+We welcome your contributions to README Template. This doc describes the process to contribute patches to README Template and the general guidelines we expect contributors to follow.
+Before You Start
+
+We accept patches in the form of github pull requests. If you are new to github, please read How to create github pull requests first.
+Contributor License Agreements
+
+Contributions to this project must be accompanied by a Contributor License Agreement. You (or your employer) retain the copyright to your contribution, this simply gives us permission to use and redistribute your contributions as part of the project.
```
Comment (eberlyn) [inline Contributing.MD]: Need to check on this

Comment (eberlyn) [reply Contributing.MD]: I do not think we should require CLAs. They are not recommended by the open source community, as they require burden on the contributor, and the norm in the open-source community is the understanding that contributions will be incorporated into the master code under the same license as the master code (also called inbound=outbound).

GitHub even explictly refers to inbound=outbound in their Terms: https://help.github.com/en/github/site-policy/github-terms-of-service#6-contributions-under-repository-license 

Basically, it is common knowledge by contributing to a repo that your contribution will be incorporated into the code under the existing license. No need to have contributors jump through extra hoops to determine this.

Recommend we incorporate language to the effect of "Contributions will be release in accordance with GitHub site policies (link to https://help.github.com/en/github/site-policy/github-terms-of-service#6-contributions-under-repository-license), including that contributions will be released under the same license applied to this repository.

Comment (eberlyn) [reply Contributing.MD]: Way to do your homework and come up with creative solutions though - gives us more to think about for sure!

File: Contributing.MD [hunk 2]
```diff
@@ -0,0 +1,30 @@
+Contributing to README Template
+
+We welcome your contributions to README Template. This doc describes the process to contribute patches to README Template and the general guidelines we expect contributors to follow.
```
Comment (eberlyn) [inline Contributing.MD]: I like how [Code.gov's CONTRIBUTING file ](https://github.com/GSA/code-gov-front-end/blob/master/CONTRIBUTING.md) lays out multiple ways users can submit issues, including creating GitHub Issues, emailing their email inbox, and creating pull requests. I think we should add something like that in here.

File: Contributing.MD [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+Before You Start
+
+We accept patches in the form of github pull requests. If you are new to github, please read How to create github pull requests first.
+Contributor License Agreements
```
Comment (eberlyn) [inline Contributing.MD]: Replace this section with: 

**Licensing Status**

As noted in LICENSE, this project is in the worldwide public domain (in the public domain within the United States, and copyright and related rights in the work worldwide are waived through the CC0 1.0 Universal public domain dedication).

All contributions to this project will be released under the CC0 dedication. By submitting a pull request, you are agreeing to comply with this waiver of copyright interest.

Comment (eberlyn) [reply Contributing.MD]: Suggest we move this section to the top, after welcome.

File: Contributing.MD (hunk 2 above)
Comment (eberlyn) [reply Contributing.MD]: I also like Code.gov's Welcome! section - touches on the coding practices/community norms we encourage project teams to include in their CONTRIBUTING files:

We're so glad you're thinking about contributing to a U.S. Government open source project! If you're unsure about anything, just ask -- or submit the issue or pull request anyway. The worst that can happen is you'll be politely asked to change something. We love all friendly contributions.

[For community norms, consider adding]: This is a friendly and encouraging environment. Discriminatory, hateful, or disrespectful conduct will not be tolerated. Contributions that portray these characteristics may not be considered, and necessary reporting of inappropriate conduct will occur.

File: Contributing.MD [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+
+Contributions to this project must be accompanied by a Contributor License Agreement. You (or your employer) retain the copyright to your contribution, this simply gives us permission to use and redistribute your contributions as part of the project.
+If you are an individual writing original source code and you're sure you own the intellectual property, then you'll need to sign an individual CLA.
+If you work for a company that wants to allow you to contribute your work, then you'll need to sign a corporate CLA.
+Contributing Process
```
Comment (eberlyn) [inline Contributing.MD]: Suggest we title this section "Contributing Via Pull Request" or "Submitting a Pull Request"

File: Contributing.MD [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+
+Most pull requests should go to the master branch and the change will be included in the next major/minor version release (e.g., 3.6.0 release). If you need to include a bug fix in a patch release (e.g., 3.5.2), make sure it’s already merged to master, and then create a pull request cherry-picking the commits from master branch to the release branch (e.g., branch 3.5.x).
+For each pull request, a Public Access Implementation and Support team member will be assigned to review the pull request. For minor cleanups, the pull request may be merged right away after an initial review. For larger changes, you will likely receive multiple rounds of comments and it may take some time to complete. We will try to keep our response time within 7-days but if you don’t get any response in a few days, feel free to comment on the threads to get our attention. We also expect you to respond to our comments within a reasonable amount of time. If we don’t hear from you for 2 weeks or longer, we may close the pull request. You can still send the pull request again once you have time to work on it.
```
Comment (eberlyn) [inline Contributing.MD]: First line - Team should be capitalized in "Public Access Implementation Support team". Also, change the sentence starting "For larger changes..." to:

For larger changes, we may have some comments or clarifications that will extend the time needed to complete.

No hyphen needed for "7-days". Change "feel free to comment on the threads" to singular "thread".

File: Contributing.MD [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+Once a pull request is merged, we will take care of the rest and get it into the final release.
+Pull Request Guidelines
+
+Create small PRs that are narrowly focused on addressing a single concern. We often receive PRs that are trying to fix several things at a time, but if only one fix is considered acceptable, nothing gets merged and both author's & review's time is wasted. Create more PRs to address different concerns and everyone will be happy.
+For speculative changes, consider opening an issue and discussing it first. 
+
+Provide a good PR description as a record of what change is being made and why it was made. Link to a GitHub issue if it exists.
+Don't fix style and formatting unless you are already changing that line to address an issue. PRs with irrelevant changes won't be merged. If you do want to fix formatting or style, do that in a separate PR.
```
Comment (eberlyn) [inline Contributing.MD]: I think this is a bit confusing. It reads as if contributors can't submit style or formatting changes at first, but then says they can in a separate PR. Are we basically trying to say only submit a PR for one section?

File: Contributing.MD [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+Unless your PR is trivial, you should expect there will be reviewer comments that you'll need to address before merging. We expect you to be reasonably responsive to those comments, otherwise the PR will be closed after 2-3 weeks of inactivity.
```
Comment (eberlyn) [inline Contributing.MD]: Delete - I think this is covered above.

File: Contributing.MD [hunk 8]
```diff
@@ -0,0 +1,30 @@
...
+Maintain clean commit history and use meaningful commit messages. PRs with messy commit history are difficult to review and won't be merged. Use rebase -i upstream/master to curate your commit history and/or to bring in latest changes from master (but avoid rebasing in the middle of a code review).
+Keep your PR up to date with upstream/master (if there are merge conflicts, we can't really merge your change).
```
Comment (eberlyn) [inline Contributing.MD]: up to date -> up-to-date

take out "really" from parentheses

File: Contributing.MD [hunk 9]
```diff
@@ -0,0 +1,30 @@
...
+Once a pull request is merged, we will take care of the rest and get it into the final release.
+Pull Request Guidelines
+
+Create small PRs that are narrowly focused on addressing a single concern. We often receive PRs that are trying to fix several things at a time, but if only one fix is considered acceptable, nothing gets merged and both author's & review's time is wasted. Create more PRs to address different concerns and everyone will be happy.
+For speculative changes, consider opening an issue and discussing it first.
```
Comment (eberlyn) [inline Contributing.MD:40]: Think we should add an Issue section before the PR sections, describing when an Issue should be opened and any guidelines for submitting those.

I think Code.gov's [CONTRIBUTING file ](https://github.com/GSA/code-gov-front-end/blob/master/CONTRIBUTING.md)has a good one.

**Review by mmiddle5 — COMMENTED** [review]
Changes made!

**Comment by eberlyn** [comment]
Thank you!!

Booz | Allen | Hamilton

Eleanor Berlyn
Associate
Mobile 703-505-7160
BoozAllen.com<http://www.boozallen.com/>

**Review by eberlyn — APPROVED** [review]

**Pull request merged**